tasks/fullStack/app/controllers/reviews_controller.rb

```diff
@@ -1,44 +1,80 @@
+    reviews_query = reviews_query.where(product_id: params[:product_id]) if params[:product_id].present?
+    @reviews = reviews_query.order(created_at: :desc).limit(REVIEW_PER_PAGE)
+
+    respond_to do |format|
+      format.html
+      format.turbo_stream
     end
   end
 
+  # GET /reviews/new
+  def new
+    @review = Review.new
+  end
+
+  # POST /reviews
   def create
-    # TODO: Create reviews in background. No need to show errors (if any) to users, it's fine to skip creating the review silently when some validations fail.
+    product = Product.find_by(shop_id: review_params[:shop_id], id: review_params[:product_id])
+    @review = Review.new(review_params.except(:shop_id))
+
+    if product.present? && @review.valid?
+      tags = tags_with_default(product, review_params)
 
-    tags = tags_with_default(params)
-    Review.create(product_id: params[:product_id], body: params[:body], rating: params[:rating], reviewer_name: params[:reviewer_name], tags: tags)
+      CreateReviewWorker.perform_async(
+        review_params[:shop_id], review_params[:product_id],
+        review_params[:body], review_params[:rating], review_params[:reviewer_name], tags
+      )
 
-    flash[:notice] = 'Review is being created in background. It might take a moment to show up'
-    redirect_to action: :index, shop_id: Product.find_by(id: params[:product_id]).shop_id
+      flash[:notice] = 'Review is being created in background. It might take a moment to show up'
+      redirect_to reviews_path(format: :html, shop_id: review_params[:shop_id]), status: :see_other
+    else
+      render "reviews/new", status: :unprocessable_entity
+    end
+  end
+
+  # GET /reviews/ratings_change?shop_id=&months=
+  def ratings_change
+    months = (params[:months] || 3).to_i
+    shop = Shop.find_by_id(params[:shop_id]) if params[:shop_id].present?
+
+    render json: { ratings_change: shop&.ratings_change(months) }
   end
 
   private
 
+  def review_params
+    params.require(:review).permit(:shop_id, :product_id, :body, :rating, :reviewer_name, :tags)
+  end
+
   # Prepend `params[:tags]` with tags of the shop (if present) or DEFAULT_TAGS
   # For simplicity, let's skip the frontend for `tags`, and just assume frontend can somehow magically send to backend `params[:tags]` as a comma-separated string
   # The logic/requirement of tags is that:
   #  - A review can have `tags` (for simplicity, tags are just an array of strings)
   #  - If the shop has some `tags`, those tags of the shop should be part of the review's `tags`
   #  - Else (if the shop doesn't have any `tags`), the default tags (in constant `DEFAULT_TAGS`) should be part of the review's `tags`
   # One may wonder what an odd logic and lenthy comment, thus may suspect something hidden here, an easter egg perhaps.
-  def tags_with_default(params)
-    product = Product.find_by(id: params[:product_id])
+  def tags_with_default(product, params)
     default_tags = product.shop.tags || DEFAULT_TAGS
```

Comment:
FYI, there is a hidden bug here, where constant `DEFAULT_TAGS` can be mutated via `concat` when `product.shop.tags` is empty.